src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -736,6 +738,15 @@ private int decodePlainFrame(
             final AmqpDescribedType descriptor = performative.kind();
             final AmqpServerDecoder decoder = decodersByPerformative.getOrDefault(descriptor, decodeUnknownType);
             server.decodeChannel = frameHeader.channel();
+
+            if (server.sessions.get(server.decodeChannel) != null &&
+                server.sessions.get(server.decodeChannel).sessionState == AmqpSessionState.DISCARDING &&
```

Comment:
Please hoist to `session` variable instead of repeating the lookup.

Reply:
Can I change `AmqpServer` and `AmqpSession` to package-private from private?